titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/resourcepool/ResourcePoolAssignment.java

```diff
@@ -23,19 +23,15 @@
 public class ResourcePoolAssignment {
 
     private final String resourcePoolName;
+    private String preferredResourcePoolName;
     private final String rule;
 
-    public ResourcePoolAssignment(String resourcePoolName, String rule) {
-        this.resourcePoolName = resourcePoolName;
-        this.rule = rule;
-    }
-
-    public String getResourcePoolName() {
-        return resourcePoolName;
+    public String getPreferredResourcePoolName() {
+        return preferredResourcePoolName;
     }
 
-    public String getRule() {
-        return rule;
+    public void setPreferredResourcePoolName(String preferredResourcePoolName) {
```

Comment:
Do not mute an object. Use builder to create a new version with this change.

Reply:
Got it. You are right the builder pattern (in this case) actually makes it easier to read.